workflow-templates/im-deploy-on-prem-website.yml

```
@@ -0,0 +1,462 @@
+
+      # TODO:  If you need to do replacements in your appsettings.json file you can do it two ways.
+      # 1- Repeat this add-or-update-json-properties action for each environment, and add an 'if: needs.set-vars.outputs.ENVIRONMENT == ""' condition with the appropriate env to each action.
+      # 2- In the set-vars job, add one set-variable-based-on-environment action block per variable then use that variable in the single add-or-update-json-properties action below.
+      # - name: Update appsettings.json with Environment Specific Values
+      #   uses: im-open/add-or-update-json-properties@v1.0.1
+      #   with:
+      #     path-to-json-file: '${{ env.PROJECT_ROOT }}/appsettings.json'
+      #     properties-to-update-or-add: |
+      #       [
+      #         {"SecretName": "SecretValue"},
+      #         {"Nested.Secret.Name": "SecretValue"},
+      #         {"": ""}
+      #       ]
+
+      # TODO:  If you need to do any Octopus variable substitution (i.e. replacing #{OCTO_PLACEHOLDER} in files) use the following action.  Must be done once per file.
+      # - uses: im-open/octostache-action@v1.0.0
+      #   with:
+      #     variablesFile: '' # Files containing the variable substitutions to make, you will need to create this file
+      #     templateFile: ''  # The file to make the substitutions in
+      #     outputFile: ''    # The file to save the substitutions in (generally the same as above)
+
+      # TODO:  If you are using any nuget/npm packages from GitHub Packages uncomment this step
+      # TODO:  If the project contains a local nuget.config, this action will not work properly.  Either:
+      #        - Remove the file and use this action or
+      #        - Delete this action and modify the file so it has the appropriate org entries in nuget.config. https://github.com/im-practices/git-er-done/blob/main/packages/connect-nuget.md#script-output
+      # - name: Authenticate with GitHub Packages
+      #   uses: im-open/authenticate-with-gh-package-registries@v1.0.1
+      #   with:
+      #     read-pkg-token: ${{ secrets.READ_PKG_TOKEN }} # This is an org level secret
+      #     orgs: '' # TODO:  Add a csv list of organizations that npm/nuget packages will be pulled from like 'im-enrollment,im-client'
+
+      # TODO:  If config transformation is needed for different environments include environment name
+      # For example to perform transformations on a web.config using the web.production.config file:
+      # dotnet publish --configuration Release /p:EnvironmentName=Production
+      - name: Build and Publish
+        working-directory: ${{ env.PROJECT_ROOT }}
+        run: |
+          dotnet build --configuration Release
+          dotnet publish -c Release -o ./published_app --no-restore
+      - name: Zip the published app for faster deployment and uploads
+        working-directory: ${{ env.PROJECT_ROOT }}
+        run: |
+          (cd published_app && zip -r ../${{ env.DEPLOY_ZIP }} .)
```

Pre-built artifact change:
```suggestion
      - name: Zip the published app for faster deployment and uploads
        working-directory: ${{ env.UNZIPPED_ASSET }}
        run: zip -r ../${{env.DEPLOY_ZIP}} .
```